lib/logger.js

```diff
@@ -1,8 +1,10 @@
-const isEnvTest =
-  typeof process !== 'undefined' && typeof process.env !== 'undefined' && process.env.BABEL_ENV === 'test'
+const isEnvTestOrNotDevelopment =
+  typeof process !== 'undefined' &&
+  typeof process.env !== 'undefined' &&
+  (process.env.BABEL_ENV === 'test' || process.env.NODE_ENV !== 'development')
```

Comment:
Just to clarify things up, in the issue's example you wrote, the warn message appears when `NODE_ENV === 'development'` but here, it looks like we're suppressing the warn message when `NODE_ENV === 'development'`.

Shouldn't we rewrite this function as such:

```javascript
const isEnvTestOrProduction =
  typeof process !== 'undefined' &&
  typeof process.env !== 'undefined' &&
  (process.env.BABEL_ENV === 'test' || process.env.NODE_ENV === 'production')
```

And then, yes: `... && !isEnvTestOrProduction`

Reply:
Thanks, I've updated the PR to reflect this feedback.